src/tl/tl/tlOptional.cc

```diff
@@ -0,0 +1,32 @@
+/*
+
+  KLayout Layout Viewer
+  Copyright (C) 2006-2024 Matthias Koefferlein
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+*/
+
+
+#include "tlOptional.h"
+
+namespace tl
+{
+
+#if __cplusplus < 201703L
```

Comment:
Why not simply using the poor man's implementation always?

Reply:
Counter question: Why using std::string and not a poor man implementation of it. :o)
And the next inclined reader will ask: Why isn't std::optional used when available? :o)

Now serious: I prefer not to re-invent something if it's already there.